presto-common/src/main/java/com/facebook/presto/common/type/UserDefinedDistinctType.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import com.facebook.presto.common.QualifiedObjectName;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class UserDefinedDistinctType
+{
+    private final QualifiedObjectName name;
+    private final TypeSignature base;
+    private final Optional<QualifiedObjectName> superTypeName;
+    private Optional<UserDefinedDistinctType> superType;
```

Comment:
Can we merge these two into one? if yo have the super type, then you already have the name. BTW it might be good to call it `parentType`. `Super` sounds parent or above.

Reply:
We don't always have the whole info. It could be we just have the super type name, but not its whole definition

Reply:
Check my other comment (https://github.com/prestodb/presto/pull/17301#discussion_r808605185). Probably name is good enough. If we are to fetch the parent type definition, then we only need to check if the child base and parent base are the same. If the are not the same, throw an error. Otherwise, saving the parent name would be good enough.

Reply:
Previously I thought parent types can have different orderability trait. But @alpaker confirmed it is not the case. So now we can just store name of parents. See the new renamed file DistinctTypeInfo